Sienna/tutorial_1.jl

```diff
@@ -0,0 +1,186 @@
+using PowerSystems
+using PowerSimulations
+using HydroPowerSimulations
+using PowerSystemCaseBuilder
+# solver
+using HiGHS
+# julia base
+using Dates
+using SHA
+using Logging
+
+#=
+    Command line argument parsing
+=#
+
+function print_help()
+    cases = readdir(joinpath(@__DIR__, "cases"); sort = false)
+    valid_cases = filter(c -> isdir(joinpath(@__DIR__, "cases", c)), cases)
+    print(
+        """
+        usage: julia --project=GenX GenX/main.jl \
+            --case=<case> | --all \
+            [--help]      \
+            [--run]       \
+            [--write]     \
+
+        ## Arguments
+
+         * `--case`:  the option of network and horizon to run. Valid cases are
+            * $(join(vec(["$(net)-$(hor)" for hor in horizon_options(), net in network_options()]), "\n    * "))
+         * `--all`    if passed, `--case` must not be passed, and the argument
+                      will loop over all valid cases
+         * `--help`   print this help message
+         * `--run`    if provided, execute the case
+         * `--write`  if provided, write out files to
+        """
+    )
+    return
+end
+
+function _parse_args(args)
+    ret = Dict{String,String}()
+    for arg in args
+        if (m = match(r"--([a-z]+)=(.+?)($|\s)", arg)) !== nothing
+            ret[m[1]] = m[2]
+        elseif (m = match(r"--([a-z]+?)($|\s)", arg)) !== nothing
+            ret[m[1]] = "true"
+        else
+            error("unsupported argument $arg")
+        end
+    end
+    return ret
+end
+
+#=
+    HiGHS overloads to print models to files
+=#
+
+const HIGHS_WRITE_FILE_PREFIX = Ref{String}("")
+
+function _write_highs_model(highs)
+    prefix = HIGHS_WRITE_FILE_PREFIX[]::String
+    if isempty(prefix)
+        return
+    end
+    instances = joinpath(dirname(@__DIR__), "instances")
+    tmp_filename = joinpath(instances, "tmp.mps")
+    HiGHS.Highs_writeModel(highs, tmp_filename)
+    # We SHA the raw file so that potential gzip differences across
+    # platforms don't matter.
+    hex = bytes2hex(open(SHA.sha256, tmp_filename))
+    run(`gzip $tmp_filename`)
+    mv(
+        "$(tmp_filename).gz",
+        joinpath(instances, "$prefix-$hex.mps.gz");
+        force = true,
+    )
+    return
+end
+
+# !!! TYPE PIRACY TO INTERCEPT ALL HIGHS SOLVES AND WRITE THEM TO FILES !!!
+function HiGHS.Highs_run(highs)
+    _write_highs_model(highs)
+    return ccall((:Highs_run, HiGHS.libhighs), Cint, (Ptr{Cvoid},), highs)
+end
+
+horizon_options() = [1, 6, 12, 24, 48]
+
+network_options() = [
+    "CopperPlate",
+    "PTDF",
+    "DC",
+    "Transport",
+]
+
+#=
+    Main Sienna loop to optimize a decision model
+=#
+
+function main(args)
+    parsed_args = _parse_args(args)
+    if get(parsed_args, "help", "false") == "true"
+        return print_help()
+    end
+    write_files = get(parsed_args, "write", "false") == "true"
+
+    # case data is downloaded from a julia artifcat
+    sys = build_system(PSISystems, "modified_RTS_GMLC_DA_sys")
```

Comment:
If you want to change the horizon beyond 48 hours use 
```transform_single_time_series(sys, Hour(HORIZON_LENGHT), Hour(24)```

Reply:
I will leave this for the future in case we need larger and harder cases